### PR TITLE
Create ORT Error and Violation Attributions

### PR DESCRIPTION
**Pull request by JakobSchubert**
For all issues in the ORT Report, generate Error attributions.

An Error attribution in the Opossum UI is:

-  a signal with follow-up and exclude-from-notice
-  comment, describing the issue
-  no coordinates, no license
-  attached to the resources that have the issue

This includes issues from analyzer and from scanner.
We try to attach the issues to the corresponding root paths. As a fallback, we attach it to the root path `/`.

### CODE REVIEW

File: reporter/src/main/kotlin/reporters/OpossumReporter.kt [hunk 1]
```diff
@@ -65,16 +75,19 @@ class OpossumReporter : Reporter {
...
-    data class OpossumSignal (
+    data class OpossumSignal(
         val source: String,
         val id: Identifier? = null,
         val url: String? = null,
         val license: SpdxExpression? = null,
         val copyright: String? = null,
         val comment: String? = null,
         val preselected: Boolean = false,
+        val followUp: String? = null,
```
Comment (sebathomas) [inline reporter/src/main/kotlin/reporters/OpossumReporter.kt:89]: We could also convert to string in the JSON converter.

File: reporter/src/main/kotlin/reporters/OpossumReporter.kt [hunk 2]
```diff
@@ -532,12 +587,22 @@ class OpossumReporter : Reporter {
             opossumInput.addPackagesThatAreRootless(analyzerResultPackages)
         }
 
+        analyzerResult.issues.entries.forEach {
+            val identifier = it.key
+            val issues = it.value
+            issues.forEach { issue ->
+                opossumInput.addIssue(issue, identifier, "ORT-Analyzer-Issues")
+            }
+
```
Comment (sebathomas) [inline reporter/src/main/kotlin/reporters/OpossumReporter.kt:596]: Unnecessary empty line

File: reporter/src/main/kotlin/reporters/OpossumReporter.kt [hunk 3]
```diff
@@ -532,12 +587,22 @@ class OpossumReporter : Reporter {
...
+        }
+
         val scannerResults = ortResult.scanner?.results?.scanResults ?: return OpossumInput()
         scannerResults.entries.forEachIndexed { index, entry ->
             log.debug("scannerResult ${index + 1} of ${scannerResults.entries.size}")
             opossumInput.addScannerResults(entry.key, entry.value, maxDepth)
         }
 
+
```
Comment (sebathomas) [inline reporter/src/main/kotlin/reporters/OpossumReporter.kt:605]: Not sure about this one either

**Review by sebathomas — APPROVED** [review]
Looking good, just some minor points.

File: reporter/src/main/kotlin/reporters/OpossumReporter.kt [hunk 4]
```diff
@@ -532,12 +587,22 @@ class OpossumReporter : Reporter {
...
         // val advisorResult = ortResult.advisor?.result
         // val evaluatorResult = ortResult.evaluator?.result
```
Comment (antonbauhofer) [inline reporter/src/main/kotlin/reporters/OpossumReporter.kt:607]: Can we remove these?

**Review by antonbauhofer — APPROVED** [review]
Looks good to me.

File: reporter/src/main/kotlin/reporters/OpossumReporter.kt (hunk 1 above)
Comment (JakobSchubert) [reply reporter/src/main/kotlin/reporters/OpossumReporter.kt:89]: Good idea, I will do that.

File: reporter/src/main/kotlin/reporters/OpossumReporter.kt (hunk 4 above)
Comment (JakobSchubert) [reply reporter/src/main/kotlin/reporters/OpossumReporter.kt:607]: Yeah, good idea. I will replace them with a TODO comment.

File: reporter/src/test/kotlin/reporters/OpossumReporterTest.kt [hunk 5]
```diff
@@ -159,11 +158,42 @@ class OpossumReporterTest : WordSpec({
...
         }
+
+        "ortIssues should exist and contain all issues" {
+
+            val issuesFromFirstPackage =
+                opossumInput.getSignalsForFile("/pom.xml/compile/first-package-group/first-package@0.0.1")
+                    .filter { it.comment?.contains(Regex("Source-.*Message-")) == true }
+            issuesFromFirstPackage.size shouldBe 4
+            issuesFromFirstPackage.forEach {
+<<<<<<< HEAD
```
Comment (maxhbr) [inline reporter/src/test/kotlin/reporters/OpossumReporterTest.kt:169]: !!

Comment (JakobSchubert) [reply reporter/src/test/kotlin/reporters/OpossumReporterTest.kt]: I fixed it.

**Review by sebathomas — APPROVED** [review]

**Pull request merged**